crowbar_framework/app/models/service_object.rb

```diff
@@ -1180,6 +1184,7 @@ def apply_role(role, inst, in_queue)
       elems.each do |role_name|
         old_nodes = old_elements[role_name]
         new_nodes = new_elements[role_name]
+        leftover_nodes = databag["deployment"][@bc_name]["elements"]["#{role_name}_remove"]
```

Comment:
Shouldn't we add stuff to `leftover_nodes`, since there can be multiple roles and we don't want to override the value?